Source/Task/LocklessList.h

```diff
@@ -0,0 +1,365 @@
+// Copyright (c) Microsoft Corporation
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#pragma once
+
+#pragma warning(push)
+#pragma warning(disable: 4324) // structure was padded due to alignas
+
+/*****************************************************************************
+ 
+ LocklessList is a lock free singly linked list that can take any type as a payload.
+ For a given payload type, it defines a type "Node" that represents a node in
+ the linked list.  Node contains your payload as a "data" member and an atomic
+ uintptr_t next value.  Note that extra data is stored in the low nibble of next so
+ it is not directly castable to the next node pointer.
+ 
+ LocklessList allocates its own node pointers, but you may also pass them as
+ parameters.
+ 
+ Typical Usage:
+ 
+ struct MyPayload
+ {
+ uint32_t someData;
+ // etc...
+ };
+ 
+ LocklessList<MyPayload> list;
+ 
+ MyPayload p = new MyPayload;
+ p->someData = 16;
+ list.push_back(p);
+ 
+ p = list.pop_front();
+ if (p != nullptr)
+ {
+ printf("Value: %d\r\n", p->data.someData);
+ delete p;
+ }
+
+ LocklessList is based on the the algorithm described in this paper:
+
+ Simple, Fast, and Practical Non-Blocking and Blocking Concurrent Queue Algorithms
+
+ Maged M. Michael Michael L. Scott
+ Department of Computer Science University of Rochester Rochester, NY 14627-0226 fmichael,scottg@cs.rochester.ed
+
+ http://cs.rochester.edu/~scott/papers/1996_PODC_queues.pdf
+ 
+ ******************************************************************************/
+
+template <typename TData>
+class alignas(sizeof(std::uintptr_t) * 8) LocklessList
```

Comment:
does this build in vs2015?
do we still care about 2015 support?

Reply:
It built for me.